### PR TITLE
minor: Add more proc-macro tests for parsing negative literals

### PR DESCRIPTION
**Pull request by Veykril**
cc https://github.com/rust-lang/rust-analyzer/pull/19371

### CODE REVIEW

File: crates/proc-macro-srv/src/tests/mod.rs [hunk 1]
```diff
@@ -136,23 +238,41 @@ fn test_fn_like_mk_literals() {
...
               LITERAL Float 3.14f64 42:2@0..100#ROOT2024
+              LITERAL Float -3.14f64 42:2@0..100#ROOT2024
               LITERAL Float 3.14 42:2@0..100#ROOT2024
+              LITERAL Float -3.14 42:2@0..100#ROOT2024
               LITERAL Integer 123i64 42:2@0..100#ROOT2024
-              LITERAL Integer 123 42:2@0..100#ROOT2024"#]],
+              LITERAL Integer -123i64 42:2@0..100#ROOT2024
+              LITERAL Integer 123 42:2@0..100#ROOT2024
+              LITERAL Integer -123 42:2@0..100#ROOT2024"#]],
     );
```
Comment (Veykril) [inline crates/proc-macro-srv/src/tests/mod.rs:276]: This output is bugged, the `-` is joined to the literal which is incorrect in our token tree model
